### PR TITLE
Clear React build warnings in client

### LINKED ISSUE #344: Clean the code, removing all the unused code for the client on the project
We are blocked by the warning of unused variables on the client now, need someone to remove all of them to let us pass the CI on the Github Action!
```
./src/components/TreeDetails.js
32
  Line 1:17:  'Component' is defined but never used                                                                      no-unused-vars
33
  Line 2:8:   'PropTypes' is defined but never used                                                                      no-unused-vars
```

### PR DESCRIPTION
**Pull request by nmcharlton**
Resolves #344 

- Removed unused dependencies
- Fixed useEffect hook dependencies

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -74,7 +73,8 @@
     "eslint-plugin-prettier": "^3.1.3",
     "eslint-plugin-react": "^7.19.0",
     "prettier": "2.0.5",
-    "react-inspector": "^3.0.2"
+    "react-inspector": "^3.0.2",
+    "typescript": "^4.0.2"
```
Comment (dadiorchen) [inline client/package.json]: We are using typescript now?

**Review by dadiorchen — APPROVED** [review]
I have no question about this PR

File: client/package.json (hunk 1 above)
Comment (nmcharlton) [reply client/package.json]: I think this must have been a to resolve a peer dependency warning from npm.
We're not using it directly, and everything installs fine without it, so I'll remove it.

Comment (nmcharlton) [reply client/package.json]: I think this must have been a to resolve a peer dependency warning from npm.
We're not using it directly, and everything installs fine without it, so I'll remove it.

**Pull request merged**